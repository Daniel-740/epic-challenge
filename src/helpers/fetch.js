
export const findById = (arrayMain, id) => arrayMain.filter( data => data.id === id);

export const checkRelationCameraType = (arrayMain, id) => {
     const response = arrayMain.find(data => data.cameraType === id);

     return response;
}