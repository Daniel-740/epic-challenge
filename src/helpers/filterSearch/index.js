/*
 *
 * @param {string} value: Is string for the value search bar
 * @param {array} arrayFilter: Is array original values for the state redux
 * @param {string} entity: to differentiate the module from the browser, this makes it reusable
 */

export const filterSearch = (value, arrayFilter, entity) => {

     if(entity === 'cameraTypes'){

          const filteredRows = arrayFilter.filter((row) => {
               return row.name.toLowerCase().includes(value.toLowerCase()) || row.description.toLowerCase().includes(value.toLowerCase());
             });

          return filteredRows
     }
/* 
     if(entity === 'cameras'){

     } */

}