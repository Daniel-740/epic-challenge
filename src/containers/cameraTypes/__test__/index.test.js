import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { CameraTypes } from '..';
import {renderWithProviders } from '../../../helpers/test'

test('render CameraTypes container', () => {

     const  component = renderWithProviders(<CameraTypes />)
     expect(component).toMatchSnapshot();
})