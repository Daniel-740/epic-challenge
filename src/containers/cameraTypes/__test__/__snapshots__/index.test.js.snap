// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CameraTypes container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root css-10ydfi0-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-17xat2w-MuiGrid-root"
        >
          <a
            href="/camera-type"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add Camera Type
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
      <div
        class="MuiContainer-root css-ur2jdm-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-15g2oxy"
        >
          <div
            class="MuiPaper-root ForwardRef-root-1 mb2 mt2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="ForwardRef-searchContainer-7"
            >
              <div
                class="MuiInputBase-root MuiInput-root ForwardRef-input-6 MuiInputBase-fullWidth MuiInput-fullWidth"
              >
                <input
                  class="MuiInputBase-input MuiInput-input"
                  placeholder="Search..."
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-searchIconButton-4"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root ForwardRef-icon-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-iconButtonHidden-3"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root ForwardRef-icon-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiTableContainer-root"
          >
            <table
              aria-label="simple table"
              class="MuiTable-root"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Id
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Description
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Actions
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root css-10ydfi0-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-17xat2w-MuiGrid-root"
      >
        <a
          href="/camera-type"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add Camera Type
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
    <div
      class="MuiContainer-root css-ur2jdm-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-15g2oxy"
      >
        <div
          class="MuiPaper-root ForwardRef-root-1 mb2 mt2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="ForwardRef-searchContainer-7"
          >
            <div
              class="MuiInputBase-root MuiInput-root ForwardRef-input-6 MuiInputBase-fullWidth MuiInput-fullWidth"
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                placeholder="Search..."
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-searchIconButton-4"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root ForwardRef-icon-5"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-iconButtonHidden-3"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root ForwardRef-icon-5"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiTableContainer-root"
        >
          <table
            aria-label="simple table"
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Id
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Description
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
